### DISCUSSION TITLE
Integration with BehaviorTree.ROS2

### DISCUSSION THREAD

**Discussion by marip8** [Yaskawa-Global/motoros2#323] [Show and tell]
I recently created a library of behavior tree node plugins for interacting with the MotoROS2 driver using [BehaviorTree.ROS2](https://github.com/BehaviorTree/BehaviorTree.ROS2). The repository can be found here:

https://github.com/marip8/motoros2_behavior_tree

You might find this useful for building a ROS2 application around a Yaskawa robot, such as [this application](https://github.com/ros-industrial-consortium/snp_automate_2023) based on [`scan_n_plan_workshop`](https://github.com/ros-industrial-consortium/scan_n_plan_workshop). Contributions and feedback welcome!

**Comment by gavanderhoorn** [comment]
Really nice work @marip8 👍 

Would you happen to have a smaller(-ish) example tree perhaps? Might make it easier for potential users to figure out how to integrate / sequence the states.

**Reply by marip8** [reply]
## Wait for digital input sub-tree
![image](https://github.com/user-attachments/assets/1a625a2c-8c1b-49d8-b07a-792415e03010)

## Use IO to start a grinding spindle device
![image](https://github.com/user-attachments/assets/9d41c4fa-8887-4bc3-b62c-94bb6012da33)

## Executing motion with trajectory mode
![image](https://github.com/user-attachments/assets/381d5c0e-763b-4828-940a-eb268f5e744a)

## Executing motion with point queue mode
![image](https://github.com/user-attachments/assets/2d301f2d-c37f-46dd-b6d0-b309a6f2f668)
